### PR TITLE
[WIP]Admin interface locale breaks  "Expiration date" field

### LINKED ISSUE #32497: Admin interface locale breaks  "Expiration date" field
### Preconditions (*)
1. Magento version 2.4.2

### Steps to reproduce (*)
1. sigh in to admin panel
1. go to System -> All users
1. change locale of current user to Any except 'en_Us'
1. try to create a new user, and set expiration date 
1. save user
1. get an error "Expiration Date is not valid"

### Expected result (*)
1. expect to create user

### Actual result (*)
1. !https://user-images.githubusercontent.com/33631491/111330717-df91bd80-8678-11eb-91c4-824bccdf455c.jpeg!

## 
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- < > Severity: **S2** *- Affects non-critical data or functionality and forces users to employ a workaround.*

@magento give me 2.4-develop instance

### PR DESCRIPTION
**Pull request by engcom-Kilo**
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/32497

### Manual testing scenarios (*)

1. see https://github.com/magento/magento2/issues/32497

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Kilo** [comment]
@magento run all tests

File: dev/tests/integration/testsuite/Magento/Security/Model/ResourceModel/UserExpirationTest.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ */
+class UserExpirationTest extends TestCase
+{
+    /**
+     * @var UserExpiration
+     */
+    private $userExpirationResource;
+
+    /**
+     * @inheridoc
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Security/Model/ResourceModel/UserExpirationTest.php:29]: @inheritDoc

File: dev/tests/integration/testsuite/Magento/Security/Model/ResourceModel/UserExpirationTest.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+        self::assertEquals($expireDate, $timeZone->formatDateTime($userExpiration->getExpiresAt()));
+    }
+
+    /**
+     * Create and save user.
+     *
+     * @return int
+     */
+    private function getUser(): int
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Security/Model/ResourceModel/UserExpirationTest.php:65]: Please use existing fixture from
`dev/tests/integration/testsuite/Magento/User/_files`
or create new one

File: dev/tests/integration/testsuite/Magento/Security/Model/UserExpiration/ValidatorTest.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+     */
+    private $localeResolver;
+
+    /**
+     * @var TimezoneInterface
+     */
+    private $timeZone;
+
+    /**
+     * @inheridoc
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Security/Model/UserExpiration/ValidatorTest.php:32]: @inheritDoc

File: dev/tests/integration/testsuite/Magento/Security/Model/UserExpiration/ValidatorTest.php [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        $this->timeZone = Bootstrap::getObjectManager()->get(TimezoneInterface::class);
+    }
+
+    /**
+     * Verify validation is passed on non US locale for date expiration.
+     *
+     * @magentoAppArea adminhtml
+     * @return void
+     */
+    public function testValidateUserExpiresAtWithNonUsLocale(): void
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Security/Model/UserExpiration/ValidatorTest.php:46]: It looks like it should be unit test.

**Review by zakdma — CHANGES_REQUESTED** [review]

File: dev/tests/integration/testsuite/Magento/Security/Model/UserExpiration/ValidatorTest.php (hunk 4 above)
Comment (engcom-Kilo) [reply dev/tests/integration/testsuite/Magento/Security/Model/UserExpiration/ValidatorTest.php:46]: Test interacts with system configuration and there should be a lot of mock objects created to make it unit. I believe it's better to leave it as integration.

File: dev/tests/integration/testsuite/Magento/Security/Model/UserExpiration/ValidatorTest.php (hunk 3 above)
Comment (engcom-Kilo) [reply dev/tests/integration/testsuite/Magento/Security/Model/UserExpiration/ValidatorTest.php:32]: Done

File: dev/tests/integration/testsuite/Magento/Security/Model/ResourceModel/UserExpirationTest.php (hunk 2 above)
Comment (engcom-Kilo) [reply dev/tests/integration/testsuite/Magento/Security/Model/ResourceModel/UserExpirationTest.php:65]: Done

File: dev/tests/integration/testsuite/Magento/Security/Model/ResourceModel/UserExpirationTest.php (hunk 1 above)
Comment (engcom-Kilo) [reply dev/tests/integration/testsuite/Magento/Security/Model/ResourceModel/UserExpirationTest.php:29]: Done

File: dev/tests/integration/testsuite/Magento/Security/Model/ResourceModel/UserExpirationTest.php (hunk 2 above)
Comment (engcom-Kilo) [reply dev/tests/integration/testsuite/Magento/Security/Model/ResourceModel/UserExpirationTest.php:65]: Done

**Pull request closed without merging**